aws/ondemand/facebook/hhvm/enter-container.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# This runs on each login to the EC2 instance, but can also be run manually.
+
+retry() {
+  "$@" 2>/dev/null && return 0
+  sleep 1
+  "$@" 2>/dev/null && return 0
+  sleep 5
+  "$@"
+}
+
+SRC_DIR=~/.ondemand
+WORK_DIR=~/.od-hhvm
+LOCK_FILE="$WORK_DIR/lock.pid"
+DOCKER_CONFIG="$WORK_DIR/docker.conf"
+
+mkdir -p "$WORK_DIR"
+
+# Prevent race conditions and other bad states.
+if ! (groups | grep docker >/dev/null); then
+  echo "Current user is not in the 'docker' group."
+  echo "Please verify if OnDemand initialization has finished successfully."
+  echo "You'll need to log out and log back in after initialization is finished."
+  exit 1
+fi
+
+LOCK_PID=$(cat "$LOCK_FILE" 2>/dev/null)
+if [ -n "$LOCK_PID" ]; then
+  if ps -p "$LOCK_PID" >/dev/null; then
+    echo "A previous instance of \`enter-container\` is still running."
+    echo "Please wait for it to finish or kill it."
+    exit 1
+  else
+    # Previous instance didn't finish cleanly and possibly left things in a bad
+    # state. Clean up and start from scratch.
+    rm -f "$DOCKER_CONFIG" "$LOCK_FILE"
+  fi
+fi
+echo $$ > $LOCK_FILE
+
+# Select a Docker image to use.
+echo "Looking up available Docker container images..."
+echo
+echo "If you don't want to open a Docker container, you can ^C at any point;"
+echo "then run \`enter-container\` if you change your mind."
+echo
+
+aws configure set default.region us-west-2
+
+if [ -e $DOCKER_CONFIG ]; then
+  source $DOCKER_CONFIG
+fi
+
+if [ -z "$IMAGE" ]; then
+  FAILED=($(
+    retry aws ecr list-images --query "imageIds[*].imageTag" --output text \
+      --repository-name hhvm-failed-builds
+  ))
+  SUCCESSFUL=($(
+    retry aws ecr list-images --query "imageIds[*].imageTag" --output text \
+      --repository-name hhvm-successful-builds
+  ))
+  echo "================================"
+  echo "Choose a Docker container image."
+  echo "--------------------------------"
+  echo
+  echo "Failed builds:"
+  printf '  %s\n' "${FAILED[@]}" | sort -r
+  echo
+  echo "Successful builds:"
+  printf '  %s\n' "${SUCCESSFUL[@]}" | sort -r
+  echo
+fi
+
+while [ -z "$IMAGE" ]; do
+  REPO=""
+  read -rp "Your choice: " CHOICE
+  echo
+  printf '%s\n' "${FAILED[@]}" | grep -Fx "$CHOICE" >/dev/null \
+    && REPO=hhvm-failed-builds
+  printf '%s\n' "${SUCCESSFUL[@]}" | grep -Fx "$CHOICE" >/dev/null \
+    && REPO=hhvm-successful-builds
+  if [ -n "$REPO" ]; then
+    IMAGE="$REPO:$CHOICE"
+    CONTAINER=""
+    rm -f $DOCKER_CONFIG 2>/dev/null
+    echo "IMAGE=$IMAGE" > $DOCKER_CONFIG
```

Comment:
a comment saying that settingm IMAGE is equivalent to `break` or similar would be nice